src/Take.Blip.Client/BlipClientBuilder.cs

```diff
@@ -236,6 +237,14 @@ public BlipClientBuilder WithPresenceStatus(PresenceStatus presenceStatus)
             return this;
         }
 
+        public BlipClientBuilder WithEnvelopeBufferSize(int envelopeBufferSize)
+        {
+            if (envelopeBufferSize < -1) throw new ArgumentOutOfRangeException(nameof(envelopeBufferSize));
+
+            EnvelopeBufferSize = envelopeBufferSize;
```

Comment:
Not sure, but i think envelope buffer size can be negative, which defines a unbounded capacity

Reply:
You're right. I'm using a blip client builder with a envelope size of -1, which means a unbounded capacity.
I wasn't sure if any negative value works. 
Should I leave this whithout any validation? What if envelope buffer size is equal to zero?

Reply:
I changed it according to the lime protocol implementation

https://github.com/takenet/lime-csharp/blob/master/src/Lime.Protocol/Client/ClientChannelBuilder.cs